<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <title>Document</title>
  </head>
  <body></body>
  <script>
    function fn1() {
      return new Promise((resolve, reject) => {
        reject();
      });
    }
    fn1()
      .then(
        function () {
          console.log("第一个成功了");
        },
        function () {
          console.log("第一个失败了"); //
        }
      )
      .then(
        function () {
          console.log("第二个成功了"); //
        },
        function () {
          console.log("第二个失败了");
        }
      )
      .then(
        function () {
          return new Promise((resolve, reject) => {
            //
            reject();
          });
        },
        function () {
          return new Promise((resolve, reject) => {
            resolve();
          });
        }
      )
      .then(
        function () {
          console.log("第四个成功了");
        },
        function () {
          console.log("第四个失败了"); //
        }
      )
      .then(
        function () {
          console.log("第五个成功了"); //
        },
        function () {
          console.log("第五个失败了");
        }
      );
    // promise对象的状态只会影响第一个then()，之后链式调用的then()都默认为执行resolve()
  </script>
  <html></html>
</html>
